incubator/hnc/api/v1alpha1/hierarchical_namespace.go

```diff
@@ -43,6 +44,10 @@ type HierarchicalNamespaceStatus struct {
 	//
 	// - "conflict": a namespace of the same name already exists. If the validation webhook
 	// works properly, this should never happen since the HNS won't be created at the first place.
+	//
+	// - "error": there's something wrong with the HNS. Currently, it would be that
```

Comment:
Maybe make this more specific - eg `"forbidden": the HNS was created in a namespace that doesn't allow children, such as kube-system or hnc-system`.

We don't have a human-readable field so we should be as specific as possible IMO.